### DISCUSSION TITLE
`onConnect` callback support

### DISCUSSION THREAD

**Discussion by Dhaiwat10** [rainbow-me/rainbowkit#407] [Ideas]
It would be really nice to have support for an `onConnect` callback which gets fired after the user connects their wallet. `wagmi` already supports this:

```tsx
const { connect } = useConnect({
    onConnect: async () => console.log('This will run when the user connects'),
  });
```
  
Not sure how to make it work with Rainbowkit though. The code snippet above only works if you use the `connect` function imported from that hook to make the connection, and not with Rainbowkit.

**Comment by Dhaiwat10** [comment]
fwiw, happy to put in a PR for this if you're interested. I need this! :)

**Comment by kasparkallas** [comment]
Can't you just use the wagmi's callback? Because RainbowKit is just interacting with wagmi anyways. I could be wrong because I haven't tried it.

**Reply by Dhaiwat10** [reply]
I tried it. Doesn't work. In order for the wagmi callback to work you need to be using the `connectors` imported from the hook like this:

```tsx
const { connectors } = useConnect({ onConnect: async () => { })

return (
	{
		connectors.map(() => <Connector />)
	}
)
```

**Reply by Dhaiwat10** [reply]
You need to be interacting with the very instances of the connectors imported from the `useConnect` hook you are passing the callback to, if you know what I mean. Rainbowkit probably creates its own instances of wagmi connectors.

**Reply by Dhaiwat10** [reply]
https://github.com/rainbow-me/rainbowkit/blob/main/packages/rainbowkit/src/wallets/useWalletConnectors.ts#L16

would probably need to pass in an `onConnect` callback here ^

**Comment by kasparkallas** [comment]
I'll also point out the fact that you can get a similar behaviour by using `useEffect` with `const { activeChain } = useNetwork();` with `activeChain` as the hook dependency. I could be wrong again...

**Comment by koolamusic** [comment]
This might interest y'all https://github.com/rainbow-me/rainbowkit/pull/389#discussion_r890169259
